src/Microsoft.AspNetCore.Server.Kestrel.Https/HttpsConnectionFilter.cs

```diff
@@ -98,9 +98,9 @@ public HttpsConnectionFilter(HttpsConnectionFilterOptions options, IConnectionFi
                 {
                     previousPrepareRequest?.Invoke(features);
 
-                    if (clientCertificate != null)
+                    if (certificateHandle != IntPtr.Zero)
                     {
-                        features.Set<ITlsConnectionFeature>(new TlsConnectionFeature { ClientCertificate = clientCertificate });
```

Comment:
Could this be changed to the following?

    var clientCertificate = ConvertToX509Certificate2(sslStream.RemoteCertificate);
    if (clientCertificate != null)
    {
        features.Set<ITlsConnectionFeature>(new TlsConnectionFeature { ClientCertificate = clientCertificate });
    }

    // ...

    private X509Certificate2 ConvertToX509Certificate2(X509Certificate certificate)
    {
        if (certificate == null)
        {
            return null;
        }

        X509Certificate2 certificate2 = certificate as X509Certificate2;
        if (certificate2 != null)
        {
            return certificate2;
        }

    #if NETSTANDARD1_3
        // conversion X509Certificate to X509Certificate2 not supported
        // https://github.com/dotnet/corefx/issues/4510
        return null;
    #else
        return new X509Certificate2(certificate);
    #endif
    }

This way we avoid copying `clientCertificate` from the `userCertificateValidationCallback` which I think is cleaner. We can also reuse the `ConvertToX509Certificate2` method in the `userCertificateValidationCallback`.